src/api/apps/articles/model/save.coffee

```diff
@@ -17,11 +18,21 @@ artsyXapp = require('artsy-xapp')
 { sanitizeLink } = require "./sanitize.js"
 chalk = require 'chalk'
 { cloneDeep } = require 'lodash'
+Channel = require '../../channels/model'
 
 @onPublish = (article, cb) =>
   unless article.published_at or article.scheduled_publish_at
     article.published_at = new Date
-  @generateSlugs article, cb
+
+  self = @
+  @generateSlugs article, (err, article) =>
+    if process.env.ENABLE_PUBLISH_RABBITMQ_EVENTS == 'true'
```

Comment:
**suggestion**: I feel like it's getting a bit confusing to have this behavior injected here in the middle of the publishing process. As is, this code would run (and the message emitted) after slugs are generated but _before_ the article is saved.

What do you think of moving this logic up [here](https://github.com/artsy/positron/blob/5d313c74b5ff4dcbfd1c32f10ddc0402f8706e4f/src/api/apps/articles/model/index.js#L166) so that we can chain this behavior after the modified article is persisted:

```javascript
if (publishing) {
  return onPublish(modifiedArticle, (err, article) => {
    // at this point, all of the publishing modifications have been set but not persisted
    sanitizeAndSave((err, article) => {
      if (err) {
        // if an error occurs here, we would not want to emit the message
        return callback(err)
      } else {
        // successfully persisted. emit message.
      }
    })(err, article)
  })
}
```

Reply:
We might also be able to leverage the ArticleModel to utilize some of its helper methods:

```javascript
model = new ArticleModel(cloneDeep article)
model.dispatchPublishEvent()
```

```coffeescript
module.exports = class Article extends Backbone.Model

  dispatchPublishEvent: ->
    if @isVisibleToPublic() and ENABLE_PUBLISH_RABBITMQ_EVENTS == 'true'
      payload =
        id: @get('id')
        featured_artist_ids: @get('primary_featured_artist_ids')
        href: @href()
        title: @get('title')

      amqp.publish("editorial", "article.published", payload)
      
```

Reply:
nice suggestion, thanks